collections/_posts/2022-11-28-CurriculumLearningByDIH.md

```diff
@@ -0,0 +1,117 @@
+
+$$
+r_{t+1}(i) = \left\{
+  \begin{array}
+    1 \gamma*a_t(i) + (1-\gamma)*r_{t}(i) \quad if \quad i \in S_t \\
+    r_t(i) \quad else
+  \end{array}
+  \right.
+$$
+
+$$ \gamma \in [0,1] $$ is a discount factor and $$S_t$$ is the set of samples used for training at time $$t$$.
+
+The three instantaneous hardness used to calculate DIH and compared are:
+* the loss : $$l(y_i,F(x_i; w_t))$$, where $$l(·, ·)$$ is a standard loss function and $$F(·; w)$$ is the model where $$w$$ are the model parameters
+* the loss change between two consecutive time steps : $$ \|l(y_i,F(x_i; w_t))-l(y_i,F(x_i; w_{t-1}))\| $$
+* the prediction flip (the 0-1 indicator of whether the prediction correctness changes) between two consecutive time steps : $$ \| 𝟙[\hat{y}^t_i = y_i]- 𝟙[\hat{y}^{t-1}_i = y_i]\| $$, where $$\hat{y}^t_i$$ is the prediction of sample $$i$$ in step $$t$$, e.g., $$ \arg\max_j F(x_i; w_t)[j]$$ for classification.
+
+![](/collections/images/CL_DIH/DIH_vs_SPL.jpg)
+
+Compared to DIH, instantaneous hardness is i) much less stable during training and ii) requires extra training and inference steps by a model over all the samples.
+
+
+## Properties
+
+DIH can vary a lot between different samples facilitating their discrimination, and so helping to select the most informative samples. When it is smaller, the samples are more memorable, i.e. easy to learn, while a larger DIH means that the samples are harder to retain.
+
+![](/collections/images/CL_DIH/sample_separation.jpg)
+The the large steps in the curves are explained by a cyclic learning rate. We clearly see the gap between the training dynamics of red samples that are harder and blue samples that have smaller DIH. Moreover, the variance of samples with larger DIH is bigger showing that a local minima is not found for them. It is thus necessary to revisit them often, and the same conclusion can be made regarding the prediction flip that is higher for the same samples.
+
+![](/collections/images/CL_DIH/early_information.jpg)
+
+DIH in early epochs (40) suffices to differentiate between easy and hard samples. It doesn't require a full training to classify the samples. 
+DIH metrics decrease during training for both easy and hard samples, indicating that as learning progresses samples become less informative, and resulting in fewer samples being needed to continue training the model.
+
+# Experiments
+
+## Curriculum learning
+
+Since the DIH represents the hardness of samples ,it seems natural to base some curriculum learning on it (DIHCL). The authors keep training the model on samples whith large DIH that have historically been hard, since the model does not perform well on them, and revisit easy samples less frequently, because the model is more likely to stay at those samples’ minima. 
```

Comment:
Misplaced comma:
```suggestion
Since the DIH represents the hardness of samples, it seems natural to base some curriculum learning on it (DIHCL). The authors keep training the model on samples whith large DIH that have historically been hard, since the model does not perform well on them, and revisit easy samples less frequently, because the model is more likely to stay at those samples’ minima. 
```